lib/strarray.c

```diff
@@ -67,7 +67,7 @@ EXPORTED void strarray_fini(strarray_t *sa)
     sa->alloc = 0;
 }
 
-EXPORTED void strarray_free(strarray_t *sa)
+EXPORTED void strarray_free(void *sa)
```

Comment:
I don't think we should do this. It removes compile-time protection that prevents us from mistakenly passing junk into strarray_free like `&sa` instead of `sa` for example.

Rather, we should do what we do elsewhere:
```
imap/jmap_mail_query.c:    free_hash_table(&cfilter->contactgroups, (void(*)(void*))strarray_free);
```

Reply:
I'm not sure that casting a function pointer type to a different function pointer type isn't undefined behaviour tbh.  If we do it elsewhere I guess it works... but what I would actually do, if not changing `strarray_free()`, is write a local wrapper function with the right signature and call `strarray_free()` from that.

Reply:
I think this legal because you're casting the incoming argument to void* which is legal?

-fsanitize=undefined does not find it with both gcc and clang, and scan-build does not complain about it

Reply:
Ahh yeah, that makes sense, though I still think it smells bad.  I've dropped the change to strarray_free, and then made a local wrapper function rather than casting.  This isn't performance-critical code, so the extra function call overhead isn't a concern.